src/Text/Pandoc/Writers/HTML.hs

```diff
@@ -933,9 +933,11 @@ inlineToHtml opts inline = do
                         linkText <- inlineListToHtml opts txt
                         slideVariant <- gets stSlideVariant
                         let s' = case s of
-                                      '#':xs | slideVariant == RevealJsSlides
-                                        -> '#':'/':xs
-                                      _ -> s
+                                   '#':xs -> let prefix = if slideVariant == RevealJsSlides
```

Comment:
That's what I first thought as well, but then I realized the current behaviour breaks manually set links as well:

    $ pandoc --id-prefix=pp-
    # head {#myhead}

    ![](img.png){#myimg}

    [link to head](#myhead) 

    [link to img](#myimg)



    <h1 id="pp-myhead">head</h1>
    <div class="figure">
    <img src="img.png" id="pp-myimg" />

    </div>
    <p><a href="#myhead">link to head</a></p>
    <p><a href="#myimg">link to img</a></p>

That's why I figured the behaviour of `--id-prefix` should be to simply prepend it to _all_ `id`s and all links that start with `#`.  This way, someone using pandoc in a wiki or CMS doesn't have to trust the input document to not clash with `id`s on the rest of the page (users of the wiki might set `id`s manually unaware the admin uses those `id`s on the page). This behaviour sounds even more useful than having only auto-generated links be prefixed and leaving the manual links alone. I don't think the readers should be bothered with this. Unless I'm missing something...?